api/envoy/config/rbac/v3/rbac.proto

```diff
@@ -310,3 +310,26 @@ message Principal {
     Principal not_id = 8;
   }
 }
+
+// Action defines the result of allowance or denial when a request matches the matcher.
+message Action {
```

Comment:
I'm a bit confused here, is this `Action` used by an end user or by the RBAC filter internally? If it's the latter, should it be moved to some other places so that it is not user facing?

Reply:
No, it is used by users.

```yaml
name: rbac
typed_config:
  "@type": type.googleapis.com/envoy.extensions.filters.network.rbac.v3.RBAC
  matcher:
    on_no_match:
      action:
        name: action
        typed_config:
          # Action
          "@type": type.googleapis.com/envoy.config.rbac.v3.Action
          name: allow_all
          action: ALLOW
```

Reply:
> could we add the example config to the RBAC API or documentation?

Generic matching has its own [documentation](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/matching/matching_api), so I would like to add an example for RBAC, both the policy one and the matcher one. To make the patch clear, I think it would be better to submit as a different PR.

> should it be added to the general matcher actions

No, I do not think so. RBAC action is not a common action. RBAC action works specifically for RBAC filters.